internal/service/elbv2/target_group.go

```diff
@@ -1037,7 +1037,9 @@ func flattenTargetGroupStickiness(attributes []*elbv2.TargetGroupAttribute) ([]i
 				m["cookie_duration"] = duration
 			}
 		case "stickiness.app_cookie.cookie_name":
-			m["cookie_name"] = aws.StringValue(attr.Value)
+			if sType, ok := m["type"].(string); !ok || sType == "app_cookie" {
```

Comment:
this looks good but where is sType being set? are we sure that the iteration will happen in the right order?

Reply:
If I'm understanding this code correctly the stickiness attributes passed into `flattenTargetGroupStickiness` are read from the live Target Group resource via the API, so at the point they are being flattened all necessary entries are guaranteed to be in the array and are just being reformatted into a map.